sdk/keyvault/keyvault-common/src/challengeBasedAuthenticationPolicy.ts

```diff
@@ -146,16 +147,25 @@ export function createKeyVaultChallengeCallbacks(
       ? parsedChallenge.resource + "/.default"
       : parsedChallenge.scope;
 
-    if (!scope) {
-      throw new Error("Missing scope.");
+    if (scope && !disableChallengeResourceVerification) {
+      verifyChallengeResource(scope, request);
     }
 
-    if (!disableChallengeResourceVerification) {
-      verifyChallengeResource(scope, request);
+    const { error, claims: base64EncodedClaims } = parsedChallenge;
+    let claims: string | undefined = undefined;
+    if (error) {
+      logger.verbose("Received error in WWW-Authenticate header:", error);
+      if (error === "insufficient_claims" && base64EncodedClaims) {
+        claims = atob(base64EncodedClaims);
+      }
     }
 
-    const accessToken = await getAccessToken([scope], {
+    const scopes = scope ? [scope] : [];
```

Comment:
Why allow empty scopes? A token request must have at least one scope

Reply:
This was from before our offline conversation where you clarified that the scopes from the earlier challenge should be used -- this should not be happening now.